components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/PasswordOnboardExecutor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.basicauth;
+
+import java.util.Collections;
+import java.util.Map;
+import org.wso2.carbon.identity.user.registration.engine.graph.Executor;
+import org.wso2.carbon.identity.user.registration.engine.model.ExecutorResponse;
+import org.wso2.carbon.identity.user.registration.engine.model.RegistrationContext;
+import org.wso2.carbon.identity.user.registration.engine.util.Constants;
+import org.wso2.carbon.identity.user.registration.mgt.exception.RegistrationFrameworkException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.application.authenticator.basicauth.PasswordOnboardRegConstants.PASSWORD_ONBOARD_EXECUTOR;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.ExecutorStatus.STATUS_USER_INPUT_REQUIRED;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.PASSWORD;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.USERNAME_CLAIM_URI;
+
+/**
+ * This class is responsible for onboarding the password of the user.
+ */
+public class PasswordOnboardExecutor implements Executor {
+
+    public String getName() {
+
+        return PASSWORD_ONBOARD_EXECUTOR;
+    }
+
+    @Override
+    public ExecutorResponse execute(RegistrationContext context) throws RegistrationFrameworkException {
+
+        ExecutorResponse response;
+        if (context.getUserInputData().get(PASSWORD) == null) {
+            response = new ExecutorResponse(STATUS_USER_INPUT_REQUIRED);
+            response.setRequiredData(Collections.singletonList(PASSWORD));
+            return response;
+        } else {
+            // Todo enforce password policies.
```

Comment:
Let's track this with a git issue

Reply:
Tracked with https://github.com/wso2/product-is/issues/23373